src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -481,10 +483,13 @@ default boolean nextBoolean() {
     default void nextBytes(byte[] bytes) {
         int i = 0;
         int len = bytes.length;
+        Unsafe unsafe = Unsafe.getUnsafe();
         for (int words = len >> 3; words--> 0; ) {
             long rnd = nextLong();
-            for (int n = 8; n--> 0; rnd >>>= Byte.SIZE)
-                bytes[i++] = (byte)rnd;
+            if (unsafe.isBigEndian())
+                rnd = Long.reverseBytes(rnd);
+            unsafe.putLong(bytes, (long)Unsafe.ARRAY_BYTE_BASE_OFFSET + i, rnd);
```

Comment:
```suggestion
            unsafe.putLong(bytes, Unsafe.ARRAY_BYTE_BASE_OFFSET + i, nextLong(), false);
```

Reply:
@liach `putLong` doesn't seem to have such an overload.

Reply:
You can see if `putLongUnaligned`, whose java code tries to put aligned long if possible, works.

Reply:
> You can see if `putLongUnaligned`, whose java code tries to put aligned long if possible, works.

The reason I didn't use it was concerned about generating slower code on platforms that don't support unaligned accesses.